src/hooks/campaigns/useTemporaryClaimedRefsManager.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * After claiming rewards, it may take a while for the campaign service to sync data from the subgraph.
+ * So we use this hook to disable it when waiting backend to be synced.
+ *
+ * This hook should be used after claim reward transaction success.
+ */
+import { useLocalStorage } from 'react-use'
+import { useCallback, useMemo } from 'react'
+
+export default function useTemporaryClaimedRefsManager(): [string[], (claimedRefs: string[]) => void] {
+  const [claimedRefs, setClaimedRefs] = useLocalStorage<string[]>('campaign-claimed-refs', [])
```

Comment:
It would be better if we store it in redux. Bc redux will store into localstorage anyway.